samples/extensions/mesh_shader_culling/README.md

```diff
@@ -0,0 +1,198 @@
+
+[Mesh Shading For Vulkan](https://www.khronos.org/blog/mesh-shading-for-vulkan)
+[Introduction to Turing Mesh Shaders](https://developer.nvidia.com/blog/introduction-turing-mesh-shaders/)
+
+GPU manufactures have recommended best practices for 
+their hardware in setting the work group and mesh size 
+number. Further reading can be found here:
+
+* [Meshlet Size tradeoffs](https://zeux.io/2023/01/16/meshlet-size-tradeoffs/)
+* NVIDIA - [Mesh Shaders in Turing](https://on-demand.gputechconf.com/gtc-eu/2018/pdf/e8515-mesh-shaders-in-turing.pdf)
+* AMD - [Sampler feedback ultimate in Mesh shaders](https://gpuopen.com/wp-content/uploads/slides/AMD_RDNA2_DirectX12_Ultimate_SamplerFeedbackMeshShaders.pdf)
+* [Timur's blog](https://timur.hu/blog/2022/mesh-and-task-shaders)
+
+## Mesh Shader
+
+A mesh shader does not execute on the same workgroup as the task shader that emitted it if a task shader exists.  
+Many workgroups can be  One task shader workgroup can launch many mesh shader workgroups.
+mesh shader is responsible for generating vertices and primitives based on the number of meshlets determined by its task
+shader. The vertices and primitives generation process can be found in the following code:
+
+```glsl
+// Vertices:
+gl_MeshVerticesEXT[k * 4 + 0].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_0, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 1].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_1, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 2].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_2, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+gl_MeshVerticesEXT[k * 4 + 3].gl_Position = vec4(2.0 * sharedData.subDimension * unitVertex_3, 0.0f, 1.0f) + sharedData.positionTransformation + displacement;
+// Indices
+gl_PrimitiveTriangleIndicesEXT[k * 2 + 0] = unitPrimitive_0 + k * uvec3(4);
+gl_PrimitiveTriangleIndicesEXT[k * 2 + 1] = unitPrimitive_1 + k * uvec3(4);
+// Assigning the color output:
+vec3 color = vec3(1.0f, 0.0f, 0.0f) * (k + 1) / sharedData.meshletsNumber;
+outColor[k * 4 + 0] = color;
+outColor[k * 4 + 1] = color;
+outColor[k * 4 + 2] = color;
+outColor[k * 4 + 3] = color;
+```
+
+More details of meshlets generation can be found in the attached article:
+
+[Using Mesh Shaders for Professional Graphics](https://developer.nvidia.com/blog/using-mesh-shaders-for-professional-graphics/)
+
+## Per-primitive culling
+
+This sample uses a simple cull functionality from the mesh 
+shader. The intention in mesh shading is to only generate geometry that is relevant to the scene.  Thus, the correct 
+method of culling is to decide via a condition if a meshlet should be generated or not.
+
+In this sample, a circular visual zone is centered at the origin, with an adjustable radius, controlled by the gui.
+When a meshlet moves out of the visual zone, its generation process will be skipped.
```

Comment:
Sorry but this is still unclear to me from this text. Does the sample implement per-primitive or per-meshlet culling? Or maybe both?